src/pkg/packager/sources/oci.go

```diff
@@ -35,59 +35,69 @@ type OCISource struct {
 }
 
 // LoadPackage loads a package from an OCI registry.
-func (s *OCISource) LoadPackage(dst *layout.PackagePaths, unarchiveAll bool) (err error) {
+func (s *OCISource) LoadPackage(dst *layout.PackagePaths, filter filters.ComponentFilterStrategy, unarchiveAll bool) (pkg types.ZarfPackage, warnings []string, err error) {
 	ctx := context.TODO()
-	var pkg types.ZarfPackage
 	layersToPull := []ocispec.Descriptor{}
 
 	message.Debugf("Loading package from %q", s.PackageSource)
 
-	optionalComponents := helpers.StringToSlice(s.OptionalComponents)
-
 	// pull only needed layers if --confirm is set
 	if config.CommonOptions.Confirm {
-
-		layersToPull, err = s.LayersFromRequestedComponents(ctx, optionalComponents)
+		pkg, err = s.FetchZarfYAML(ctx)
+		if err != nil {
+			return pkg, nil, err
+		}
+		if filter != nil {
```

Comment:
I want to better understand why filtering is only done on confirm

Reply:
Filtering should be done on both

Reply:
fixed